### PR TITLE
IBM-277 label of 'available male/female beds' disappears

### PR DESCRIPTION
**Pull request by chrisns**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -13,9 +13,8 @@ RUN yum install -y  \
 
 RUN mkdir -p /opt/nodejs
 WORKDIR /opt/nodejs
-RUN curl https://nodejs.org/dist/v4.2.2/node-v4.2.2-linux-x64.tar.gz | tar xz --strip-components=1
+RUN curl https://nodejs.org/dist/v4.4.7/node-v4.4.7-linux-x64.tar.gz | tar xz --strip-components=1
```
Comment (cksanders) [inline Dockerfile]: Update README:

Install NodeJS 4.2 LTS **-> 4.4.7**

$ nvm install 4.2 **-> 4.4.7**
$ nvm use 4.2 **-> 4.4.7**

File: app/js/directives/CentreDirective.js [hunk 2]
```diff
@@ -9,17 +9,16 @@ function CentreDirective() {
     },
     link: function (scope, element, attrs) {
       scope.$watch('attributes.maleAvailability', function (newvalue) {
-        scope.attributes.highlightMale = newvalue < 1 ? 'highlight' : '';
-        scope.attributes.maleAvailability = newvalue === 0 ? 'FULL' : newvalue;
-        scope.attributes.statusMale = newvalue < 1 ? 'No Available Male Beds' : 'Available Male Beds';
-
-      });
```
Comment (cksanders) [inline app/js/directives/CentreDirective.js]: third argument required? 
reference inequality via the !== should be sufficient, as attributes.maleAvailability is an int and not a complex object

File: app/js/directives/CentreDirective.js [hunk 3]
```diff
@@ -9,17 +9,16 @@ function CentreDirective() {
...
-        scope.attributes.highlightFemale = newvalue < 1 ? 'highlight' : '';
-        scope.attributes.femaleAvailability = newvalue === 0 ? 'FULL' : newvalue;
-        scope.attributes.statusFemale = newvalue < 1 ? 'No Available Female Beds' : 'Available Female Beds';
+        scope.highlightFemale = newvalue < 1 ? 'highlight' : '';
+        scope.femaleAvailability = newvalue === 0 ? 'FULL' : newvalue;
+        scope.statusFemale = newvalue < 1 ? 'No Available Female Beds' : 'Available Female Beds';
 
-      });
+      }, true);
     },
```
Comment (cksanders) [inline app/js/directives/CentreDirective.js]: same as above.

File: app/js/directives/CentreDirective.js (hunk 2 above)
Comment (chrisns) [inline app/js/directives/CentreDirective.js]: yes third argument is required, apparently, it enforces deeper checking
where are you suggesting to use `!==`?

File: Dockerfile (hunk 1 above)
Comment (chrisns) [inline Dockerfile]: done

File: app/js/directives/CentreDirective.js (hunk 2 above)
Comment (cksanders) [inline app/js/directives/CentreDirective.js]: $watch(watchExpression, listener, [objectEquality]);

1. **objectEquality = false** :: Inequality is determined according to reference inequality, strict comparison via the !== Javascript operator, 

2. **objectEquality = true** :: inequality of the watchExpression is determined according to the angular.equals function. angular.equals (relevant for complex objects not integers) : 

- Both objects or values pass === comparison.

**'attributes.maleAvailability' relevant (and the equivalent of objectEquality = false)**

- Both objects or values are of the same type and all of their properties are equal by comparing them with angular.equals. 

**'attributes.maleAvailability' is not an object.** 

- Both values are NaN. (In JavaScript, NaN == NaN => false. But we consider two NaN as equal)

**'attributes.maleAvailability' will not be NaN**

- Both values represent the same regular expression (In JavaScript, /abc/ == /abc/ => false. But we consider two regular expressions as equal when their textual representation matches).

**'attributes.maleAvailability' is not a regular expression.**

File: app/js/directives/CentreDirective.js (hunk 3 above)
Comment (chrisns) [inline app/js/directives/CentreDirective.js]: fair, updated

**Pull request merged**